ogr/services/pagure.py

```diff
@@ -180,6 +180,26 @@ def exists(self):
     def is_fork(self) -> bool:
         return "fork" in self.namespace
 
+    @property
+    def parent(self) -> Optional["GitProject"]:
+        """
+        Return parent project if this project is a fork, otherwise return None
+        """
+        if self.is_fork:
+            parent = self._pagure.get_parent()
+            kwargs = self._pagure_kwargs.copy()
+            kwargs.update(
+                repo=self.repo,
+                namespace=parent["namespace"],
+                instance_url=self.instance_url,
+                token=self._token,
+            )
+            if "username" not in kwargs:
+                kwargs["username"] = self.service.user.get_username()
```

Comment:
FYI: you can squash these two lines into
```python
  kwargs.setdefault("username", self.service.user.get_username())
```

Reply:
Not my code :P I just copied it.